### ISSUE TITLE
API cannot be used twice

### ISSUE THREAD

**Issue by ExpHP** [dftbplus/dftbplus#247]
## Description

It appears that `dftbp_final` does not clean up everything.  If you try to use the API again after a call to `dftbp_final`, the program will abort during the second call to `dftb_process_input` as it tries to allocate memory that's already allocated.

## Steps to reproduce

1. Modify the **`test_initc`** tester to run twice:

```diff
diff --git a/test/api/mm/testers/test_fileinitc.c b/test/api/mm/testers/test_fileinitc.c
index e222e37..c999395 100644
--- a/test/api/mm/testers/test_fileinitc.c
+++ b/test/api/mm/testers/test_fileinitc.c
@@ -40,6 +40,8 @@ int main()
   double *gradients, *gross_charges;
   int natom;
 
+  for(int ii=0; ii<2; ii++) {
+
   /* Initialise DFTB+ input tree from input in external file */
   dftbp_init(&calculator, NULL);
   dftbp_get_input_from_file(&calculator, "dftb_in.hsd", &input);
@@ -71,6 +73,7 @@ int main()
   /*  clean up */
   dftbp_final(&calculator);
 
+  }
 
   /* Save some data for the internal test system */
   dftbp_write_autotest_tag(NR_OF_ATOMS, 0, mermin_energy, gradients, gross_charges, NULL);
```

Notice that `dftbp_final` is correctly called before the second call to `dftbp_init`.

2. Do **`make test_api`.**

`test_initc` will report `Incomplete`.

3. Look at **`_build/test/api/mm/testcases/fileinitc/stderror.log`**

You will find the following error message, written right before the program aborted.

```
At line 1086 of file initprogram.f90
Fortran runtime error: Attempting to allocate already allocated variable 'kpoint'
Command exited with non-zero status 2
```

**Comment by aradi** [comment]
Thanks for reporting, I'll have a look at it.

**Comment by aradi** [comment]
I think, I have fixed it in PR #259 . Should be on master in a few days. Any testing is welcome.
